pkg/api/bytes.go

```diff
@@ -90,14 +89,15 @@ func (s *Service) bytesUploadHandler(w http.ResponseWriter, r *http.Request) {
 	// Add the tag to the context
 	ctx := sctx.SetTag(r.Context(), tag)
 	p := requestPipelineFn(putter, r)
-	ctx, cancel := context.WithCancel(ctx)
+	pctx, cancel := context.WithCancel(ctx)
 	defer cancel()
-	pr := ioutil.TimeoutReader(ctx, r.Body, 5*time.Minute, func(n uint64) {
-		logger.Error(nil, "idle read timeout exceeded")
-		logger.Debug("idle read timeout exceeded", "bytes_read", n)
+	timer := time.AfterFunc(5*time.Minute, func() {
```

Comment:
This limits the whole upload operation to 5 minutes.
There could be direct uploads take could take longer than that.

Reply:
I am confused here, it was one minute before for idle reads.
Now, we are limiting the whole upload operation to 5 minutes, right?

Reply:
The problem has never been reading, but uploading.

Reply:
I agree with @istae. This doesnt seem right. The timeout was for idle read, so it meant that every chunk had 5 mins to being pushed. Now we are limiting the whole upload operation to 5 mins. This will not be enough for larger uploads.

I think we should bring back the timeout reader, just make it work properly.

Reply:
We can even remove the whole timeout thing, really.
Afaik we added it to stop requests from hanging because chunks were not going through, at this point, i don't think it adds to much value.

Reply:
> I agree with @istae. This doesnt seem right. The timeout was for idle read, so it meant that every chunk had 5 mins to being pushed. Now we are limiting the whole upload operation to 5 mins. This will not be enough for larger uploads.

The timeout reader never worked properly and acted as `time.AfterFunc` (the same as this PR).

> I think we should bring back the timeout reader, just make it work properly.

If we make the timeout reader work properly, it wouldn't fix this PR #2946, since the read was never and always finished. The problematic part was/is the upload.